### DISCUSSION TITLE
[DISCUSS] Unity Catalog glossary/terminology

### DISCUSSION THREAD

**Discussion by MrPowers** [unitycatalog/unitycatalog#212] [Ideas]
Some recommended terms:

* A "Unity Catalog instance" consists of many catalogs
* A "Unity Catalog schema" consists of many "Unity Catalog data assets"
* A Delta table, Parquet table, JSON volume, "Unity Catalog function", or Lance volume are examples of "Unity Catalog data assets"
* A Delta table can be registered as a "managed Delta table" or "external Delta table" in a "Unity Catalog schema"
* A "Unity Catalog schema" is different than a "Delta table schema".  The standalone word "schema" should not be used because it's ambiguous.

Let me know what you think about these terms.  Any other terms we should formally define?

**Comment by newfront** [comment]
The phrase: a "schema" contains one or more "tables" each with its own "schema" - has bugged me when describing catalogs too.

Contextualized words are tricky. 

The catalog has many schemas. Each schema acts like a logical bucket/organizational unit. There is a relationship between the current catalog, and the tables through the schema....

Can't we just go back to thinking catalogs have a database that in turn has tables and views? Or what about catalogs are laid out creating hierarchical organizational groups using three namespaces.

<catalog_name>.<group_node>.<table_a>

--------

**Reply by zen-data** [reply]
if we reuse the iceberg terminology, then `namespace` is another nice way to bridge the gap between catalogs and collections of tables.

**Comment by avriiil** [comment]
Similar to "schema" there is the concept of "metadata":
-- UC has metadata about data assets (path, date_created, etc.) --> "catalog metadata"
-- Delta Lake / Iceberg tables have metadata -->  "table metadata"

**Comment by avriiil** [comment]
`Metastore` is another term that seems important to define properly:
- A metastore is the top-level container for metadata in Unity Catalog. It stores metadata about data assets and the permissions that govern access to them.

**Reply by MrPowers** [reply]
Ah yes, "metastore" 🙃 

There are a lot of different definitions of "metastore".  We should definitely define this term unambiguously.

**Comment by everettVT** [comment]
The "schema" term has been distracting for me as well, though I'm not sure theres much hope of altering it given its precedence with Postgres. 

Iceberg on the other hand refers to schemas in the same way I think we are discussing, where we are defining the column names and types.
